### PR TITLE
GODRIVER-2287 Add section to troubleshooting FAQ per driver with top SEO results

### PR DESCRIPTION
**Pull request by prestonvasquez**
GODRIVER-2287

We are adding a `docs` directory at the top level of the Go Driver project. [Looking through trending go projects](https://github.com/trending/go) it's pretty noticeable that `docs` is a fairly constant practice, I have only seen a few instances of `doc` (although one of these includes [the standard library](https://cs.opensource.google/go/go)) and one instance of `documentation` . Within our organization, many of the drivers have some variation of a docs directory:

[mongo-ruby-driver](https://github.com/mongodb/mongo-ruby-driver/tree/master/docs)
[mongo-rust-driver](https://github.com/mongodb/mongo-rust-driver/tree/main/docs)
[mongo-node-driver](https://github.com/mongodb/node-mongodb-native/tree/main/docs)
[mongo-python-driver](https://github.com/mongodb/mongo-python-driver/tree/master/doc)
[mongo-cxx-driver](https://github.com/mongodb/mongo-cxx-driver/tree/master/docs)
[mongo-csharp-driver](https://github.com/mongodb/mongo-csharp-driver/tree/master/Docs)

The team has decided to stick with markdown over RST. Naming the FAQ file from this context `common-issues` matches what is being done in the [python](https://github.com/mongodb/mongo-python-driver/blob/master/doc/common-issues.rst) and [ruby](https://github.com/mongodb/mongo-ruby-driver/blob/master/docs/tutorials/common-errors.txt) drivers. 

Of the [six issues proposed ](https://jira.mongodb.org/browse/GODRIVER-2287?focusedCommentId=4352275&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4352275), five of them were unique:

GODRIVER-1763
GODRIVER-854
GODRIVER-1721
GODRIVER-1772
GODRIVER-627

GODRIVER-1763 and GODRIVER-1772 are both unresolved, GODRIVER-1721 is more of a drivers questions than a Go Drivers question. Hence, of the five unique SEO proposals, I believe only two of them are valid for Go Driver documentation: 

GODRIVER-854
GODRIVER-627

Along with the `common-issues.md`, I've also included the `CONTRIBUTING.md` in the docs directory.

### CODE REVIEW

**Comment by prestonvasquez** [comment]
Do we want to link files from the docs directory in the top level README.md file?

**Review by matthewdale — COMMENTED** [review]
The new common issues doc looks great!

Please update the link to `CONTRIBUTING.md` in the main `README.md` and consider adding a link from the `README.md `to the new `docs/common-issues.md` doc.

File: docs/common-issues.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+jsonBytes, err := bson.MarshalExtJSON(doc, true, false)
+if err != nil {
+	log.Fatalf("error encoding json: %v", err)
+}
+
+m := make(map[string]interface{})
+if err := json.Unmarshal(jsonBytes, &m); err != nil {
+	log.Fatalf("error decoding json: %v", err)
+}
+fmt.Printf("json: %+v\n", m)
```
Comment (benjirewis) [inline docs/common-issues.md:49]: What does `+v` do? I've never seen that before.

Comment (prestonvasquez) [reply docs/common-issues.md:49]: This actually isn't needed here, it's just something I do habitually, but it's used to print the fields of a struct with the values.

**Review by benjirewis — APPROVED** [review]
Great job!

**Pull request merged**